### ISSUE TITLE
2.x deprecation tracker

### ISSUE THREAD

**Issue by tafsiri** [tensorflow/tfjs#3232]
Tracking issue for things being deprecated in the 2.x release and for things that were deprecated in 1.x being removed.

### deprecations from 1.x

**tfjs-core**

- [x] batchNormalization
- [x] batchNormalization2d
- [x] batchNormalization3d
- [x] batchNormalization4d
- [x] interface MetaGraphInfo
- [x] interface SignatureDefInfo
- [x] interface SavedModelTensorInfo

**tfjs-data**

- [x] dataset.forEach
- [x] type DataElement

**tfjs-node**

- [x] createTypeOpAttr

### deprecations being added in 2.x

Add @deprecated tag and deprecationWarn runtime call

 - [x] addStrict
 - [x] subStrict
 - [x] powStrict
 - [x] mulStrict
 - [x] divStrict
 - [x] modStrict
 - [x] minimumStrict
 - [x] maximumStrict
 - [x] squaredDifferenceStrict
 - [x] notEqualStrict
 - [x] lessStrict
 - [x] equalStrict
 - [x] lessEqualStrict
 - [x] greaterStrict
 - [x] greaterEqualStrict
 - [x] setWasmPath (tfjs-backend-wasm)

### non api based breaking changes
 - [x] auto registration of gradients in tfjs-core

**Comment by tafsiri** [comment]
cc @annxingyuan if you do end up deprecating a wasm method, could you add it to this list.

**Comment by annxingyuan** [comment]
@tafsiri done
